sycl/include/sycl/accessor.hpp

```diff
@@ -2929,6 +2961,14 @@ class __SYCL_EBO host_accessor
   }
 
 public:
+  using value_type = DataT;
+
+  using iterator = typename detail::accessor_iterator<value_type, Dimensions>;
```

Comment:
`host_accessor` is inherited from `accessor` that has all those member types, why do we need to explicitly list them here again?

Reply:
They weren't picked up for some reason and hence caused the failure in the `iterator-member-types` test, whereas adding them fixed the test. Could these be better implemented by explicitly defining them using AccessorT, e.g. `using iterator = AccessorT::iterator;`?

Reply:
> host_accessor is inherited from accessor that has all those member types

Do we expect that to take place after the deprecated access-modes will finally be deleted?

Reply:
> > host_accessor is inherited from accessor that has all those member types
> 
> Do we expect that to take place after the deprecated access-modes will finally be deleted?

I'm not familiar enough with the inheritance scheme and motivation for it to answer this. Perhaps @romanovvlad can answer here?

Note that if we go this way, we would have to duplicate implementations of all common methods like `size`, `begin`, etc. And we may be prone to have the same bugs: this code defines `value_type` as `DataT` always, which is not entirely correct. From [4.7.6.10.1. Interface for buffer host accessors](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_interface_for_buffer_host_accessors):
```
using value_type = // const DataT for read-only accessors, DataT otherwise
      __value_type__;
```

> They weren't picked up for some reason and hence caused the failure in the iterator-member-types test

I'm curious what could have caused that. I tried to replicate a part of your change to `host_accessor` class, i.e. adding `IsAccessReadOnly` member and new methods and `check-sycl` passed for me (I used `AccessorT::value_type` instead of defining it explicitly)

Reply:
>> Do we expect that to take place after the deprecated access-modes will finally be deleted?


>> I'm not familiar enough with the inheritance scheme and motivation for it to answer this. Perhaps @romanovvlad can answer here?

Not sure how it's connected to deprecated access-modes removal.

Reply:
I removed those lines and used `typename AccessorT::value_type` instead and it resolved that issue and others so went with that implementation instead.